src/dataModels/JSON.js

```diff
@@ -537,39 +536,9 @@ var JSON = DataModel.extend('dataModels.JSON', {
         return this.isDrillDown() && showTree;
     },
 
-
-    /**
-     * @memberOf dataModels.JSON.prototype
-     * @param {number} colIndex
-     * @param keys
-     */
-    toggleSort: function(colIndex, keys) {
-        this.incrementSortState(colIndex, keys);
-        this.applyAnalytics({columnSort: true});
-    },
-
-    /**
-     * @memberOf dataModels.JSON.prototype
-     * @param {number} columnIndex
-     * @param {boolean} deferred
-     */
-    unSortColumn: function(columnIndex, deferred) {
-        var sorts = this.getSortedColumnIndexes(),
-            sortPosition;
-
-        if (sorts.find(function(sortSpec, index) {
-                sortPosition = index;
-                return sortSpec.columnIndex === columnIndex;
-            })) {
-            sorts.splice(sortPosition, 1);
-            if (!deferred) {
-                this.applyAnalytics({columnSort: true});
-            }
-        }
-    },
-
     /**
      * @memberOf dataModels.JSON.prototype
+     * @desc returns the columns that currently sorted and their intended direction of the sort
      */
     getSortedColumnIndexes: function() {
```

Comment:
I was just thinking something like filter and sort state were agnostic or the dataSource/app logic -> setData called that handles it. So they wouldn't be apart of any mixin but the grid table state?

I wonder what filter state would look like? Maybe a string of the characters supplied for filtering for all columns